app/components/NewUserForm.tsx

```diff
@@ -0,0 +1,113 @@
+import { useToken } from "@/utils/auth";
+import Form from "react-bootstrap/Form";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+
+type NewUserInputs = {
+  name: string;
+  email: string;
+  app_metadata: { roles: string[] };
+};
+
+export default function NewUserForm() {
+  const token = useToken();
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<NewUserInputs>({
```

Comment:
react-hook-form with typescript is super easy—just pass in the type and suddenly the whole form is typesafe 🔥